src/api/Api.cpp

```diff
@@ -4,33 +4,91 @@
+  result->ok = true;
+  if(isRegistered()) {
+    Logger::debug("[ApiClass::registerStation()] Station already registered");
+    result->message = "Station already registered";
+    return result;
+  }
+  
   accessToken = String("");
   accessTokenMillis = 0;
   if (Config::getRegistratonToken().equals("")) {
     Config::setRegistrationState(Config::RegistrationState::REGISTRATION_ERROR);
     Logger::debug("[ApiClass::registerStation()] Registration fail - no registration token");
-    return false;
+    result->message = "No registration token";
+    return result;
   }
 
   if (!Internet::isConnected()) {
     Config::setRegistrationState(Config::RegistrationState::REGISTRATION_ERROR);
     Logger::debug("[ApiClass::registerStation()] Registration fail - no internet connection");
-    return false;
+    result->message = "No internet connection";
+    return result;
+  }
+
+  Config::RegistrationState currentRegistrationState = Config::getRegistrationState();
+
+  if(currentRegistrationState < Config::RegistrationState::REGISTERED) {
+    if(!doRegister(result)) result;
+  }
+
+  if(currentRegistrationState < Config::RegistrationState::STATION_NAME) {
+    if(!doStationName(result)) result;
   }
-  Config::setRegistrationState(Config::RegistrationState::REGISTERING);
 
+  if(currentRegistrationState < Config::RegistrationState::STATION_ADDRESS) {
+    if(!doStationAddress(result)) result;
+  }
+
+  if(currentRegistrationState < Config::RegistrationState::STATION_LOCATION) {
+    if(!doStationLocation(result)) result;
+  }
+
+  if(currentRegistrationState < Config::RegistrationState::TEMP_SENSOR) {
+    if(!doTempSensor(result)) result;
+  }
+
+  if(currentRegistrationState < Config::RegistrationState::HUMIDITY_SENSOR) {
+    if(!doHumiditySensor(result)) result;
+  }
+  
+  if(currentRegistrationState < Config::RegistrationState::PREASSURE_SENSOR) {
+    if(!doPreasurreSensor(result)) result;
+  }
+  
+  if(currentRegistrationState < Config::RegistrationState::PM1_SENSOR) {
+    if(!doPM1Sensor(result)) result;
+  }
+
+  if(currentRegistrationState < Config::RegistrationState::PM2_5_SENSOR) {
+    if(!doPM2_5Sensor(result)) result;
+  }
+
+  if(currentRegistrationState < Config::RegistrationState::PM10_SENSOR) {
+    if(!doPM10Sensor(result)) result;
+  }
+
+  Logger::info("[ApiClass::registerStation()] Registered successfull");
+  Config::setRegistrationState(Config::RegistrationState::REGISTERED_OK);
+  Config::save();
+
+  return result;
+}
+
+bool ApiClass::doRegister(RegistrationResult* result) {
   String registrationToken = Config::getRegistratonToken();
   String apiUrlBase = Config::getApiUrl();
   String url = apiUrlBase + "/auth/register-station";
 
-  DynamicJsonDocument doc(JSON_OBJECT_SIZE(1));
+  DynamicJsonDocument doc(JSON_OBJECT_SIZE(2));
   Logger::debug(registrationToken.c_str());
   doc["stationRegistrationToken"] = registrationToken.c_str();
+  doc["mac"] = Bluetooth::getMAC();
```

Comment:
Shouldn't be "macAddress" like in API docs?
```suggestion
  doc["macAddress"] = Bluetooth::getMAC();
```